### PR TITLE
[full-ci] Notification bell badge count

### PR DESCRIPTION
**Pull request by lookacat**
## Description
Implement notification badge counter for bell icon.
Set
`notifications: {
    'ocs-endpoints': ['list', 'get', 'delete']
 }`

## Screenshots (if appropriate):
Screenshot (31)
Screenshot (30)

Animation:
[screen-capture.webm](https://user-images.githubusercontent.com/786551/219460758-911fe2db-b73b-4411-8b0d-b16ce3e7f8c5.webm)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-runtime/src/components/Topbar/NotificationBell.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    const notificationCountLabel = ref(`${props.notificationCount}`)
+    const animate = ref(false)
+
+    const updateNotificationLabel = () => {
+      if (props.notificationCount > 99) {
+        notificationCountLabel.value = '99+'
+      } else {
+        notificationCountLabel.value = `${props.notificationCount}`
+      }
+    }
```
Comment (JammingBen) [inline packages/web-runtime/src/components/Topbar/NotificationBell.vue:47]: I think you can make your life much easier by just creating a computed prop for the notification label that reads `props.notificationCount` (and adds `+` if needed). It should handle all the reactive stuff automatically (if I'm not overseeing something 😅 ).

Comment (lookacat) [reply packages/web-runtime/src/components/Topbar/NotificationBell.vue:47]: oh yeah right, i think i had an 'brett vorm kopf' :D

File: packages/web-runtime/src/components/Topbar/NotificationBell.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  },
+  setup(props) {
+    const { $gettext } = useGettext()
+    const notificationsLabel = computed(() => $gettext('Notifications'))
+    const notificationCountLabel = ref(`${props.notificationCount}`)
+    const animate = ref(false)
+
+    const updateNotificationLabel = () => {
+      if (props.notificationCount > 99) {
+        notificationCountLabel.value = '99+'
```
Comment (JanAckermann) [inline packages/web-runtime/src/components/Topbar/NotificationBell.vue:43]: Maybe early return instead of else ?

**Review by JanAckermann — APPROVED** [review]
LGTM, just a minor thing but that's personal preference

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/32954/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/32954/tracing/copy-and-move-resources-in-personal-space-alice-2023-2-17-02-52-22.zip 


</p></details>

**Review by JammingBen — APPROVED** [review]

File: packages/web-runtime/src/components/Topbar/Notifications.vue [hunk 3]
```diff
@@ -1,18 +1,6 @@
...
-      variation="inverse"
-      :aria-label="notificationsLabel"
-    >
-      <oc-icon
-        class="oc-cursor-pointer oc-flex oc-flex-middle"
-        name="notification-3"
-        fill-type="line"
-      />
-    </oc-button>
+    <notification-bell />
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/Notifications.vue:3]: You forgot to set the `notification-count` prop

File: packages/web-runtime/src/components/Topbar/Notifications.vue [hunk 4]
```diff
@@ -70,8 +58,12 @@
...
 import { eventBus } from 'web-pkg/src/services/eventBus'
 import { ShareStatus } from 'web-client/src/helpers/share'
+import NotificationBell from './NotificationBell.vue'
 
 export default {
+  components: {
+    NotificationBell
+  },
   computed: {
     ...mapGetters(['activeNotifications', 'configuration']),
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/Notifications.vue:69]: The `notificationLabel` computed prop in line 70 is unused now (you duplicated it to the NotificationBell component). Needs to be removed here.

File: packages/web-runtime/src/components/Topbar/Notifications.vue [hunk 5]
```diff
@@ -70,8 +58,12 @@
...
```
Comment (kulmann) [inline packages/web-runtime/src/components/Topbar/Notifications.vue:68]: `configuration` getter is unused, please remove (not touched by you, but still a good chance to clean up)

File: packages/web-runtime/tests/unit/components/Topbar/NotificationBell.spec.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    })
+    await wrapper.setProps({ notificationCount: input })
+    expect(wrapper.find('.badge').text()).toBe(expected)
+  })
+})
+function getWrapper({ mountType = mount, capabilities = {}, mocks = {} } = {}) {
+  const localMocks = { ...defaultComponentMocks(), ...mocks }
+  const storeOptions = {
+    ...defaultStoreMockOptions,
+    capabilities: () => capabilities
```
Comment (kulmann) [inline packages/web-runtime/tests/unit/components/Topbar/NotificationBell.spec.ts:47]: capabilities are not relevant for the `NotificationBell` component. You can remove them here from the storeOptions, from the options object in the `getWrapper` signature and from the tests above.

**Review by kulmann — DISMISSED** [review]
Some nitpicks, see comments. Component looks nifty, including the animation. Really nice!

**Pull request merged**